### ISSUE TITLE
User input is not sanitize before making a request resulting to send all contents of db.json file.

### ISSUE THREAD

**Issue by reny-pacheco** [Njong392/Abbreve#128]
### Description

User input is not sanitize before making a request.
Adding `../db` as form input results to send the whole content of `db.json` file.

I suggest to sanitize user input before sending a request.

### Screenshots

![image](https://user-images.githubusercontent.com/64537418/195387767-65a8d6ae-1f2e-4c57-9fea-11cac2f3afca.png)

**Comment by reny-pacheco** [comment]
Yes, I like to work on this. Please assign to me.

**Comment by Njong392** [comment]
Sure thing. Assigned!

**Comment by reny-pacheco** [comment]
Hi @Njong392 , I already created a PR for this and its ready for review, Thanks. 😃

**Comment by reny-pacheco** [comment]
Hello @Njong392 , @mathiasayivor , in relation to this issue, I found out that special characters in abbreviation is encoded when creating its `.json` file. And requesting `s/o` doesn't return its abbreviation.

Example: `s/o` ➡️ `s%2Fo.json`

**Question:** Is it advisable to allow contributors to add special characters to abbreviation?

**Comment by mathiasayivor** [comment]
> Hello @Njong392 , @mathiasayivor , in relation to this issue, I found out that special characters in abbreviation is encoded when creating its `.json` file. And requesting `s/o` doesn't return its abbreviation.
> 
> Example: `s/o` arrow_right `s%2Fo.json`
> 
> **Question:** Is it advisable to allow contributors to add special characters to abbreviation?

Yes, we definitely have to encode abbreviations with special characters, since those characters are not allowed as file names.

And yes, developers can encode abbreviations, but they must include mapping for the encoded version in the `public/server/encodedAbbrMappings.json` file.

Also, thanks for pointing this out, as I thought searching for encoded abbreviations was added after the last breaking change (#64) was merged
Why don't you include this fix in #129?

**Comment by reny-pacheco** [comment]
> Also, thanks for pointing this out, as I thought searching for encoded abbreviations was added after the last breaking change (#64) was merged Why don't you include this fix in #129?

Based on my understanding, fetching of abbreviations goes directly to `/db/<abbrev>.json` which sometimes contain the encoded abbrev.
Where is` encodedAbbrMapppings.json` used for? since the requested abbrev comes from  `/db/<abbrev>.json`?

Also thanks for responding to my questions. 😃

**Comment by mathiasayivor** [comment]
> > Also, thanks for pointing this out, as I thought searching for encoded abbreviations was added after the last breaking change (#64) was merged Why don't you include this fix in #129?
> 
> Based on my understanding, fetching of abbreviations goes directly to `/db/<abbrev>.json` which sometimes contain the encoded abbrev. Where is` encodedAbbrMapppings.json` used for? since the requested abbrev comes from `/db/<abbrev>.json`?
> 
> Also thanks for responding to my questions. smiley

The `encodedAbbrMappings.json` is to help us easily find the original version for encoded abbreviations.
